### ISSUE TITLE
git submodule libraries

### ISSUE THREAD

**Issue by sh-dave** [lix-pm/lix.client#5]
Coming from a [Kha/khamake](https://github.com/kode/kha) background, i used git submodules for versioning. Now i'm working on a couple of projects not using it, but i'm trying lix instead. I think it would be very nice to have submodule support for that in lix as well. What i have in mind:

from project root, some-lib is already added as submodule in haxe_libraries
```
lix install submodule:haxe_libraries/some-lib
```

final directory layout
```
/project
    /haxe_libraries
        /some-lib - this is the git submodule
        some-lib.hxml - created by lix and would point to project/haxe_libraries/some-lib
```

I can now have different versions for each project without having to constantly push to git and run lix during development, and mainly using VSCode nowadays, i can even browse the lib as it lives in the project directory as well.

Thoughts?

**Comment by back2dos** [comment]
I did think about this, because both haxeshim and switchx are actually held together by submodules. In theory submodules are great and actually render all package managers redundant. In practice, the question about how to deal with updates is very tricky. You could add a post-checkout hook or something to update submodules automatically, but if there are local modifications to the submodule (for whatever reason) then that fails.

We can start of course by allowing to *add* submodules that way, but the question remains how exactly they can be *managed* from that point forward. I'm happy for any suggestions.

**Comment by sh-dave** [comment]
For now my usecases are just a couple of independent libs i can easily update to new versions by hand.
So i'm good with 'lets add them' and a 'manual updates only' hint in the docs.

As for updating, shouldn't nested submodules help with that?
```
project
    tink_http@abc
        tink_url@bcd
            tink_stringly@cde
            tink_core@xyz
        tink_io@def
            tink_streams@efg
            tink_core@xyz
```

If you have everything nested, a simple submodule update should pull in everything else.
You only need to make sure that after resolving everything with lix that only 1 version of each lib is actually used (tink_core@xyz) in the example?

**Comment by back2dos** [comment]
Oh, that's right - nested submodules. Didn't even properly think about that ^^

What I mean by "update" is really "submodule update" or whatever it's called (I'm pretty much a git noob) when you check out the actual submodule commits that the parent's commit refers to. The way haxeshim is built, if you switch between two commits for which all the libs had been downloaded already, it will compile. 

If you use something heavier like SourceTree, then it will also update all submodules - [for better or worse](https://answers.atlassian.com/questions/154204/git-submodule-update-on-branch-switch). But without that in place, the result is often a version mismatch - which I would prefer to avoid.

**Comment by sh-dave** [comment]
Hm, that depends entirely on the client you're using (i'm using smartgit and pretty noob as well :sweat_smile: ). And with smartgit, i can disable automatic submodule updates.
But yeah it would suck to break dependencies depending on what client you're using.

**Comment by sh-dave** [comment]
I think that can be closed in favor of #10
